src/node/cli.ts

```diff
@@ -667,6 +669,26 @@ export const shouldRunVsCodeCli = (args: Args): boolean => {
   return extensionRelatedArgs.some((arg) => argKeys.includes(arg))
 }
 
+/**
+ * Reads the socketPath which defaults to a temporary directory
+ * with another directory called vscode-ipc.
+ *
+ * If it can't read the path, it throws an error and returns undefined.
+ */
+export async function readSocketPath(path: string): Promise<string | undefined> {
```

Comment:
💅 Mainly stylistic changes:
- make pure by passing in path
- export so we can test
- add check for `isNodeJSErronoException` so TS knows it has `error.code` and so we only throw if not a Node.js exception.

Reply:
This is because if the file isn't found, we don't care which is why we ignore `"ENOENT"` because we create a socket later ourselves.

Reply:
Thanks for the description!

Reply:
I like how you refactored the code to be more testable 👍